poly-commitment/src/combine.rs

```diff
@@ -585,127 +586,122 @@ pub fn affine_shamir_window_table_one<P: SWModelParameters>(
     res
 }
 
-fn window_shamir<G: AffineCurve>(
-    x1: G::ScalarField,
-    g1: G,
-    x2: G::ScalarField,
-    g2: G,
-) -> G::Projective {
+fn window_shamir<G: AffineRepr>(x1: G::ScalarField, g1: G, x2: G::ScalarField, g2: G) -> G::Group {
     let [_g00_00, g01_00, g10_00, g11_00, g00_01, g01_01, g10_01, g11_01, g00_10, g01_10, g10_10, g11_10, g00_11, g01_11, g10_11, g11_11] =
         shamir_window_table(g1, g2);
 
-    let windows1 = BitIteratorBE::new(x1.into_repr()).tuples();
-    let windows2 = BitIteratorBE::new(x2.into_repr()).tuples();
+    let windows1 = BitIteratorBE::new(x1.into_bigint()).tuples();
+    let windows2 = BitIteratorBE::new(x2.into_bigint()).tuples();
 
-    let mut res = G::Projective::zero();
+    let mut res = G::Group::zero();
 
     for ((hi_1, lo_1), (hi_2, lo_2)) in windows1.zip(windows2) {
         res.double_in_place();
         res.double_in_place();
         match ((hi_1, lo_1), (hi_2, lo_2)) {
             ((false, false), (false, false)) => (),
-            ((false, true), (false, false)) => res.add_assign_mixed(&g01_00),
-            ((true, false), (false, false)) => res.add_assign_mixed(&g10_00),
-            ((true, true), (false, false)) => res.add_assign_mixed(&g11_00),
-
-            ((false, false), (false, true)) => res.add_assign_mixed(&g00_01),
-            ((false, true), (false, true)) => res.add_assign_mixed(&g01_01),
-            ((true, false), (false, true)) => res.add_assign_mixed(&g10_01),
-            ((true, true), (false, true)) => res.add_assign_mixed(&g11_01),
-
-            ((false, false), (true, false)) => res.add_assign_mixed(&g00_10),
-            ((false, true), (true, false)) => res.add_assign_mixed(&g01_10),
-            ((true, false), (true, false)) => res.add_assign_mixed(&g10_10),
-            ((true, true), (true, false)) => res.add_assign_mixed(&g11_10),
-
-            ((false, false), (true, true)) => res.add_assign_mixed(&g00_11),
-            ((false, true), (true, true)) => res.add_assign_mixed(&g01_11),
-            ((true, false), (true, true)) => res.add_assign_mixed(&g10_11),
-            ((true, true), (true, true)) => res.add_assign_mixed(&g11_11),
+            ((false, true), (false, false)) => res.add_assign(&g01_00),
```

Comment:
`res += &g01_00` would be nicer here : o

Reply:
Done.